### PR TITLE
libtool -> 2.4.7

### PR DESCRIPTION
**Pull request by saltedcoffii**
needs armv7l binaries. This version of libtool finally finally finally fixes our issue with no file command at `/usr/bin/file`

#### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING=1 CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew CREW_TESTING_BRANCH=libtool_2.4.7 crew update
```

### CODE REVIEW

File: packages/libtool.rb [hunk 1]
```diff
@@ -3,35 +3,34 @@
 class Libtool < Package
   description 'GNU libtool is a generic library support script. Libtool hides the complexity of using shared libraries behind a consistent, portable interface.'
   homepage 'https://www.gnu.org/software/libtool/'
-  version '2.4.6-4'
+  version '2.4.7'
   license 'GPL-2'
   compatibility 'all'
-  source_url 'https://ftp.gnu.org/gnu/libtool/libtool-2.4.6.tar.gz'
-  source_sha256 'e3bd4d5d3d025a36c21dd6af7ea818a2afcd4dfc1ea5a17b39d7854bcd0c06e3'
+  source_url 'https://ftpmirror.gnu.org/libtool/libtool-2.4.7.tar.xz'
```
Comment (uberhacker) [inline packages/libtool.rb:9]: Good call.

**Comment by uberhacker** [comment]
After tests ran for over 16 hours on an arm build, I ended up with the following:
```
Running tests...
  GEN      public-submodule-commit
/usr/local/bin/make  check-recursive
make[1]: Entering directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
Making check in .
make[2]: Entering directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
/usr/local/bin/make  check-local
make[3]: Entering directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
  CC       libltdl/libltdlc_la-lt_dlloader.lo
  CC       libltdl/loaders/libltdlc_la-preopen.lo
  CC       libltdl/libltdlc_la-lt__alloc.lo
  CC       libltdl/libltdlc_la-lt_error.lo
  CC       libltdl/libltdlc_la-ltdl.lo
  CC       libltdl/libltdlc_la-slist.lo
  CCLD     libltdl/libltdlc.la
## ----------------------------- ##
## GNU Libtool 2.4.7 test suite. ##
## ----------------------------- ##

Functions shared with configure and libtool.

  1: func_munge_path_list works                      ok

Libtoolize operation.

  2: libtoolize macro installation                   ok
  3: libtoolize macro directory mismatch error       ok
  4: multiple AC_CONFIG_MACRO_DIRS invocation        ok
  5: multiple AC_CONFIG_MACRO_DIRS directories       ok
  6: libtoolize ACLOCAL_AMFLAGS extraction           ok
  7: libtoolize macro serial update                  ok
  8: libtoolize config files serial update           ok
  9: diagnose missing LT_CONFIG_LTDL_DIR             ok
 10: copy ltdl.m4 with shared macro directory        ok
 11: correctly parse LTDL_INIT from configure.ac     ok
 12: diagnose missing LTDL_INIT invocation           ok
 13: upgrading verbatim style aclocal.m4             ok
 14: verbatim aclocal.m4 w/o AC_CONFIG_MACRO_DIRS    ok
 15: nonrecursive ltdl with AC_CONFIG_MACRO_DIRS     ok
 16: subproject ltdl with unconventional layout      ok
 17: Subproject ltdl without GNU M4                  ok
 18: LIBTOOLIZE_OPTIONS                              ok
 19: cleanup old installation                        ok

Basic libtool operation.

 20: check help output                               ok
 21: diagnose no mode specified                      ok
 22: quote shell meta-characters in filenames        ok
 23: transform source suffices                       ok
 24: check link mode operation                       ok
 25: check objectlist file operation                 ok
 26: test LT_SUPPORTED_TAG interface                 ok

Linking and loading.

 27: link against a preloaded static library         ok
 28: build and dynamically load a module             ok
 29: preload static and dynamic module               ok
 30: deplibs_check_method                            ok
 31: disable fast install                            ok
 32: force PIC objects                               ok
 33: force non-PIC objects                           ok
 34: hardcoding library path                         ok
 35: binary relinking at install time                ok
 36: uninstalled libraries have priority             ok
 37: linking with long file names                    ok
 38: override pic_flag at configure time             ok
 39: test --with-pic                                 skipped (with-pic.at:40)

Convenience libraries.

 40: build and link against a static library         ok
 41: build and link against a dynamic library        ok
 42: build both static and dynamic                   ok
 43: allow_undefined_flag                            ok
 44: C convenience archives                          ok
 45: C++ convenience archives                        ok
 46: F77 convenience archives                        ok
 47: FC convenience archives                         ok
 48: Java convenience archives                       skipped (convenience.at:231)

Library interdependencies.

 49: static library interdependencies                ok
 50: shared library interdependencies                ok
 51: shared and static interdependencies             ok
 52: disable fast install                            ok
 53: binary relinking at install time                ok

Testing libtool functions.

 54: standard command line options                   ok
 55: mode short-hands                                ok
 56: debug tracing                                   ok
 57: duplicate members in archive tests              ok
 58: duplicate convenience archive names             ok
 59: preserve duplicate convenience deps             skipped (duplicate_deps.at:67)
 60: passing CC flags through libtool                ok
 61: passing CXX flags through libtool               ok
 62: passing F77 flags through libtool               ok
 63: passing FC flags through libtool                ok
 64: passing GCJ flags through libtool               skipped (flags.at:24)
 65: inherited_linker_flags                          ok
 66: Link order test                                 ok
 67: Link order of deplibs                           FAILED (link-order2.at:126)
 68: Failure tests                                   ok
 69: shlibpath_overrides_runpath                     ok
 70: Runpath in libtool library files                ok
 71: static linking flags for programs               FAILED (static.at:177)
 72: ccache -all-static                              ok
 73: Export test                                     ok
 74: export from a DLL with a .def file              skipped (export-def.at:35)
 75: sys_lib_search_path                             ok
 76: sys_lib_search_path on w32                      skipped (search-path.at:84)
 77: indirect convenience                            ok
 78: indirect uninstalled                            ok
 79: static library contains static library          expected failure (archive-in-archive.at:57)
 80: both of -o prog and -o prog$EXEEXT work         skipped (exeext.at:29)
 81: execute mode                                    ok
 82: bindir basic lib test                           ok
 83: bindir install tests                            ok
 84: cwrapper for uninstalled executables            ok
 85: cwrapper string length                          skipped (cwrapper.at:190)
 86: cwrapper and installed shared libraries         ok
 87: deplib in subdir                                ok
 88: inferred tag                                    ok
 89: CXX inferred tag                                ok
 90: F77 inferred tag                                ok
 91: FC inferred tag                                 ok
 92: GCJ inferred tag                                skipped (infer-tag.at:85)
 93: localized compiler messages                     skipped (localization.at:60)
 94: nocase library search                           skipped (nocase.at:31)
 95: Install tests                                   ok
 96: versioning                                      ok

DESTDIR tests

 97: Simple DESTDIR install                          ok
 98: DESTDIR with in-package deplibs                 ok

Support for older m4 interface.

 99: AM_PROG_LIBTOOL                                 ok
100: AC_WITH_LTDL                                    ok
101: AC_CONFIG_MACRO_DIR                             ok

Libtool subdir-objects support.

102: C subdir-objects                                ok
103: C++ subdir-objects                              ok

Libltdl functionality.

104: lt_dlexit unloading libs                        ok
105: lt_dlopenadvise library loading                 ok
106: lt_dlopen error messages                        expected failure (lt_dlopen.at:102)
107: lt_dlopen archive                               ok
108: lt_dlopenext error messages                     ok
109: libdir of installed modules                     ok
110: ltdl API                                        skipped (ltdl-api.at:32)
111: dlloader API                                    ok
112: loadlibrary error messages                      skipped (loadlibrary.at:250)
113: syntax of .la files                             expected failure (lalib-syntax.at:133)
114: resident modules                                ok
115: SList functionality                             ok
116: enforced lib prefix                             ok

Dynamic ltdl runtime loading.

117: dynamically ltdl preload static modules         ok
118: dynamically ltdl load a shared module           ok
119: ltdl load shared and static modules             ok
120: ltdl dryrun                                     ok
121: link with library that loads ltdl modules       ok

Standalone Libltdl.

122: compiling softlinked libltdl                    ok
123: compiling copied libltdl                        ok
124: installable libltdl                             ok
125: linking libltdl without autotools               ok

Subproject Libltdl.

126: compiling softlinked libltdl                    ok
127: compiling copied libltdl                        ok
128: installable libltdl                             ok
129: linking libltdl without autotools               ok

Nonrecursive Automake Libltdl.

130: compiling softlinked libltdl                    ok
131: compiling copied libltdl                        ok
132: installable libltdl                             ok

Recursive Automake Libltdl.

133: compiling softlinked libltdl                    ok
134: compiling copied libltdl                        ok
135: installable libltdl                             ok

C++ language support.

136: static library                                  ok
137: shared library                                  ok
138: shared and static together                      ok
139: allow_undefined_flag                            ok

C++ template tests.

140: simple template test                            ok
141: template test with subdirs                      ok

Constructors.

142: C++ static constructors                         ok
143: C++ exception handling                          skipped (exceptions.at:361)

libtool script generation.

144: config.status                                   ok
145: config.lt                                       ok

Libtool usage in GCC

146: AC_NO_EXECUTABLES                               ok

Detecting identical deplibs.

147: build tree relpaths                             expected failure (deplibs-ident.at:69)

configure interface to libltdl.

148: installable libltdl                             ok
149: --with-ltdl-include/lib                         ok
150: --with-included-ltdl                            ok
151: convenience libltdl                             ok

F77 language support.

152: static library                                  ok
153: shared library                                  ok
154: shared and static together                      ok

Fortran language support.

155: static library                                  ok
156: shared library                                  ok
157: shared and static together                      ok

Mac OS X tests

158: darwin fat compile                              skipped (darwin.at:43)
159: darwin concurrent library extraction            ok
160: darwin gdb debug information                    skipped (darwin.at:155)
161: darwin ld warnings changing configure results   ok
162: darwin can lt_dlopen .dylib and .so files       skipped (darwin.at:250)

Windows tests.

163: dumpbin -symbols section hiding                 skipped (dumpbin-symbols.at:42)
164: deplibs without file command                    ok

sysroot tests

165: -L=.../lib -l                                   skipped (sysroot.at:204)
166: -L SYSROOT/.../lib -l                           skipped (sysroot.at:205)
167: SYSROOT/.../*.la                                skipped (sysroot.at:206)

Libtool stress test.

168: Link option thorough search test                ok
169: Run tests with low max_cmd_len                  FAILED (cmdline_wrap.at:48)

## ------------- ##
## Test results. ##
## ------------- ##

ERROR: 148 tests were run,
7 failed (4 expected failures).
21 tests were skipped.
## -------------------------- ##
## testsuite.log was created. ##
## -------------------------- ##

Please send `tests/testsuite.log' and all information you think might help:

   To: <bug-libtool@gnu.org>
   Subject: [GNU Libtool 2.4.7] testsuite: 67 71 169 failed

You may investigate any problem if you feel able to do so, in which
case the test suite provides a good starting point.  Its output may
be found below `tests/testsuite.dir'.

make[3]: *** [Makefile:2555: check-local] Error 1
make[3]: Leaving directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
make[2]: *** [Makefile:1948: check-am] Error 2
make[2]: Leaving directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
make[1]: *** [Makefile:1648: check-recursive] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/libtool.20220603074627.dir/libtool-2.4.7'
make: *** [Makefile:1950: check] Error 2
/usr/local/lib/crew/lib/package.rb:258:in `system': Command failed with exit 2: make
libtool failed to build: `CFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " CXXFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FCFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " LDFLAGS="-flto " make -j8 check` exited with 2
```

**Comment by saltedcoffii** [comment]
Can you see the logs of the specific tests that failed?

**Comment by uberhacker** [comment]
I didn't add the `-k` switch to keep the logs so, unfortunately, those were lost.

**Pull request merged**